thoth/storages/graph/postgres.py

```diff
@@ -3775,6 +3775,98 @@ def create_python_cve_record(
 
                 return False
 
+    def update_missing_flag_package_version(
+        self,
+        package_name: str,
+        package_version: str,
+        index_url: str,
+        value: bool,
+    ) -> None:
+        """Update value of is_missing flag for PythonPackageVersion."""
+        with self._session_scope() as session:
+            subq = (session.query(PythonPackageVersion)
+                    .join(PythonPackageIndex)
+                    .filter(PythonPackageVersion.package_name == package_name)
+                    .filter(PythonPackageVersion.package_version == package_version)
+                    .filter(PythonPackageIndex.url == index_url)
+                    .with_entities(PythonPackageVersion.id)
+                    )
+            (
+                session.query(PythonPackageVersion)
+                .filter(PythonPackageVersion.id.in_(subq))
+                .update({"is_missing": value}, synchronize_session='fetch')
+            )
```

Comment:
Don't we want `upsert` functionality in this case?
Have a look at https://github.com/thoth-station/storages/blob/158d7a368cc25e984c23080ff02f91cbc7a7827f/thoth/storages/graph/postgres.py#L4646

Reply:
This should never insert into the table. Not sure if that matters.

Reply:
You want to update an existing record in the database, by default that flag is Null right now I guess, but if you want to insert a value on existing record without creating a new record, you need to use upsert, or insert on conflict. Update only will create a new record as far as I'm aware of and from I understood we don't want a new record but changing a value of an existing one.

Reply:
The default value should be false, I think we set it to not be nullable in the configuration

Reply:
> The default value should be false, I think we set it to not be nullable in the configuration

Then we need the upsert instead of the update, otherwise you will create a new record in the table and not updating the existing one.